components/MapComponent.tsx

```diff
@@ -1,170 +1,127 @@
+    setCurrentMarkers([...currentMarkers, e.nativeEvent.coordinate]);
+  }*/
+  
+  const mapMarkerCreations = () => {
+  return markers.map((report) =>
+    <Marker
+    key = {report.key}
+    title = {report.place}
+    coordinate = {{ latitude: report.lat, longitude: report.lng }}
+    pinColor = {report.pinColor}
+    onPress={() => {onMarkerClick(report.lat, report.lng); setButtonVisible(true); setSelectedMarker(report)}}
+    icon={require('../assets/icon.png')}
+    >
+    </Marker >)
+  }
 
-// IDK PUT THE LOCATION YOU FETCH FROM THE LOCATION SERVICES HERE? Might be different, could just be placeholder
-const area = {
-  latitude: 34.059761,
-  longitude: -118.276802,
-  latitudeDelta: 0.0922,
-  longitudeDelta: 0.0421
-}
+  function onMarkerClick (lat: number, long: number) {
+    mapRef.current.animateToRegion({
+      latitude: lat,
+      longitude: long,
+      latitudeDelta: initialRegion.latitudeDelta - 0.02,
+      longitudeDelta: initialRegion.longitudeDelta - 0.02,
+    })
+    
+  }
 
-// Function to use map in order to go through objects and use information from them, for marker locations/information
-const mapMarkerCreations = () => {
-  return state.markers.map((report) =>
-  <Marker
-  key = {report.key}
-  title = {report.place}
-  coordinate = {{ latitude: report.lat, longitude: report.lng }}
-  >
-  </Marker >)
-}
+  function onRecenter () {
+    mapRef.current.animateToRegion({
+      latitude: initialRegion.latitude,
+      longitude: initialRegion.longitude,
+      latitudeDelta: initialRegion.latitudeDelta,
+      longitudeDelta: initialRegion.longitudeDelta,
+    })
+  }
 
-// Map view component itself
-function MapComponent () {
   return (
-    <View style={styles.container}>
+    <View style={containStyle}>
       <MapView
-          style={styles.map}
+          style={mapStyle}
           provider={PROVIDER_GOOGLE}
-          region={area}
-          rotateEnabled={false}
-          scrollEnabled={false}
+          initialRegion={initialRegion}
+          ref={mapRef}
+          rotateEnabled={movementSettings.canRotate}
+          scrollEnabled={movementSettings.canScroll}
+          followsUserLocation={true}
+          showsUserLocation={true}
+          zoomEnabled={movementSettings.canZoom}
+          customMapStyle={[
+            {
+              "featureType": "poi",
+              "stylers": [{ "visibility": "off" }]
+            },{
+              "featureType": "transit",
+              "stylers": [{ "visibility": "off" }]
+            }
+          ]}
       >
       {mapMarkerCreations()}
       </MapView>
+      <View
+        style={{
+        position: 'absolute',//use absolute position to show button on top of the map
+        top: '70%', //for center align
+        alignSelf: 'flex-end' //for align to right
+        }}
+      >
+      { 
+        buttonVisible == true &&
+        <Button title="Re-Center" onPress={() => {onRecenter(); setButtonVisible(false);}}/>
```

Comment:
One thing we should probably be conscious of in the future should be to not have these debug buttons and states directly in the production components we're working on. Rather we should essentially be able to put the component into a _debug harness_ (which can be a storybook story, or just a normal component if storybook doesn't work for whatever reason) and do our visual testing from the outside. This makes it so that we don't have to come back and remove all of this debug code later on, when the component needs to _actually_ be displayed in the app. Additionally, we're able to keep the visual test harness around so that anyone who needs to play with the UI can do so later in a non-destructive manner.

To give a simple example of this:

```tsx
// BAD - Has debug code inside the component itself
const MapComponent = (props: MapViewProps) => {
    // ...
    const recenterButtonTapped = () => {
        // ...
    }
    return (
        <View>
            {/* We're co-mingling debug logic with stuff from react-native-maps now... not good...  */}
            <MapView {...} />
            <Button title="Re-Center" onPress={recenterButtonTapped} />
       </View>
    )
}

// GOOD - Uses debug code on the outside, and keeps MapComponent separated
const MapPreview = () => {
    const mapRef = useRef(null)
   
    const recenterButtonTapped = () => {
        mapRef.recenterMap()
    }

    return return (
        <View>
            {/* Here we render the MapComponent that we made and not MapView, much better  */}
            <MapComponent ref={mapRef} />
            <Button title="Re-Center" onPress={recenterButtonTapped} />
       </View>
    )
}
```

Also FYI to not copy what I have directly, but just and idea for what this pattern may look like.